system/jlib/jlog.cpp

```diff
@@ -2391,22 +2391,25 @@ void setupContainerizedLogMsgHandler()
         if (logConfig->hasProp(logFormatAtt))
         {
             const char *logFormat = logConfig->queryProp(logFormatAtt);
-            bool newFormatDetected = false;
+            HandleLogMsgHandler *previousStderrHandler{nullptr};
             if (streq(logFormat, "xml"))
             {
+                previousStderrHandler = theStderrHandler;
                 theStderrHandler = new HandleLogMsgHandlerXML(stderr, MSGFIELD_STANDARD);
```

Comment:
unfortunately, it is not thread safe to set theStderrHandler and delete the active one like this.
There are quite a few places that call queryStderrLogMsgHandler() and could be using the active one.

Either:
1) need to ensure all uses of theStderrHandler are thread safe, e.g. by changing queryStderrLogMsgHandler to a getStderrLogMsgHandler, that links it allowing callers to continue to use it while the active one is replaced, OR
2) assert that theStderrHandler cannot change (only properties of it, like the filter can).

1) is really the correct path, but will involve a lot (albeit small) changes wherever queryStderrLogMsgHandler() is used.
2) is the pragmatic change, and I don't think we really want to change the format on the fly, which is the only reason the handler changes.